src/Fluid/Flu_Prepare.cpp

```diff
@@ -114,10 +161,24 @@ void Flu_Prepare( const int lv, const double PrepTime,
          for (int j=0; j<FLU_NXT; j++)
          for (int i=0; i<FLU_NXT; i++)
          {
-            const int t = IDX321( i, j, k, FLU_NXT, FLU_NXT );
+#           if ( ELBDM_SCHEME == ELBDM_HYBRID )
+            if ( amr->use_wave_flag[lv] ) {
+#           endif // # if ( ELBDM_SCHEME == ELBDM_HYBRID )
 
+            const int t = IDX321( i, j, k, FLU_NXT, FLU_NXT );
             for (int v=0; v<FLU_NIN; v++)    fluid[v] = h_Flu_Array_F_In[TID][v][t];
 
+#           if ( ELBDM_SCHEME == ELBDM_HYBRID )
+            } else { // if ( amr->use_wave_flag[lv] ) {
+//          convert to smaller array for HYB_GHOST_SIZE ghost zones
+            real (*smaller_h_Flu_Array_F_In   )[FLU_NIN ][CUBE(HYB_NXT)] = (real (*)[FLU_NIN][CUBE(HYB_NXT)]) h_Flu_Array_F_In;
```

Comment:
[Important] In this case, the size of the above `(i,j,k)` loops should be `HYB_NXT` instead of `FLU_NXT`.